app/src/main/java/chat/rocket/android/authentication/twofactor/presentation/TwoFAPresenter.kt

```diff
@@ -50,4 +54,15 @@ class TwoFAPresenter @Inject constructor(private val view: TwoFAView,
     fun signup() {
         navigator.toSignUp()
     }
+
+    private suspend fun registerPushToken() {
+        // TODO: put it on constructor
+        val localRepository: LocalRepository = SharedPreferencesRepository(navigator.activity)
```

Comment:
Same as on LoginPresenter, also I would create an Interactor for this, so you can inject the same interactor on both presenters and just call a single method on the Interactor.